### PR TITLE
cleanup gateway block, add descriptions

### PR DESCRIPTION
**Pull request by mrentsch65**
update with master #48

### CODE REVIEW

**Review by lorandmolnar — APPROVED** [review]

**Review by BaRu-ifm — APPROVED** [review]

File: JSON_for_IO-Link.yaml [hunk 1]
```diff
@@ -4974,6 +4974,8 @@ components:
       enum:
         - MANUAL
         - DHCP
+        - AUTO_IP
+        - DCP
```
Comment (fm-sick) [inline JSON_for_IO-Link.yaml:4978]: is DCP here at the right place ? DCP is a service and not a configuration.

Comment (fm-sick) [reply JSON_for_IO-Link.yaml:4978]: all of this was already exiting in the released specification.

Comment (fm-sick) [reply JSON_for_IO-Link.yaml:4978]: this is a subject to be changed in the future.

**Review by vervoortb — APPROVED** [review]

File: JSON_for_IO-Link.yaml (hunk 1 above)
Comment (julian-st) [reply JSON_for_IO-Link.yaml:4978]: a description for the added enum values would be good

**Pull request closed without merging**